### PR TITLE
Cross-build for Scala Native

### LINKED ISSUE #92: Consider publishing for scala-native
There has been progress in scala-native world recently. It got [runtime](https://github.com/armanbilge/epollcat) for cats-effect. As this library is built on cats-effect, would it be possible to add support for scala-native?

### PR DESCRIPTION
**Pull request by kubukoz**
Closes #92.

### CODE REVIEW

File: .sbtopts [hunk 1]
```diff
@@ -0,0 +1 @@
+-J-Xmx4G
```
Comment (kubukoz) [inline .sbtopts:1]: Linking native usually takes more than 1G, 4 was enough for me.

File: build.sbt [hunk 2]
```diff
@@ -1,13 +1,15 @@
+Global / onChangedBuildSource := ReloadOnSourceChanges
+
 val V = new {
   val cats            = "2.8.0"
   val catsEffect      = "3.3.14"
-  val circe           = "0.15.0-M1"
-  val http4s          = "0.23.15"
-  val munit           = "0.7.29"
-  val munitCatsEffect = "1.0.7"
+  val circe           = "0.14.3"
```
Comment (kubukoz) [inline build.sbt:6]: Had to downgrade due to 0.15 not having an artifact with Native 0.4 support. To be honest I'd recommend sticking to 0.14.x anyway, as it's also what http4s-circe 0.23.x (and many other libs in the ecosystem) uses.

It's only here for tests so it shouldn't be a breaking change.

**Comment by hejfelix** [comment]
could you add the native build to the `root` project like so:

```scala
lazy val root =
  project
    .in(file("."))
    .aggregate(core.jvm, core.js, core.native, http4s.jvm, http4s.js, slf4j)
    .settings(
      publish / skip := true,
    )
```

**Comment by kubukoz** [comment]
Ah, ser, I was sure I did it. I think there's a way to get all axes without enumerating them, I'll update the PR later today.

**Comment by hejfelix** [comment]
Ah. Nvm I'm thinking of scala versions. Maybe we just gotta leave out the suffix for the crossprojects?

**Comment by kubukoz** [comment]
Not sure what you mean - the platform-specific suffix gets added automatically.

Published artifacts look like this:

```
tree -L 2 ~/.ivy2/local/org.legogroup
/Users/kubukoz/.ivy2/local/org.legogroup
├── woof-core_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
├── woof-core_native0.4_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
├── woof-core_sjs1_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
├── woof-http4s_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
├── woof-http4s_native0.4_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
├── woof-http4s_sjs1_3
│   └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
└── woof-slf4j_3
    └── 0.4.6+1-819eed6c+20220926-1934-SNAPSHOT
```

**Comment by kubukoz** [comment]
@hejfelix updated to include the aggregation - it's not super convenient syntax, but at least we don't have to list out all the individual platform-specific projects.

**Comment by hejfelix** [comment]
Looks good to me now, I'll merge and cut a new release soon. Thank you very much for working on this 👌

**Pull request merged**